paper_sections/02-gams.Rmd

```diff
@@ -1,33 +1,39 @@
+Figure \ref{fig:smoothing_effect} shows optimal smoothing (where the smoothing parameter is estimated to give a parsimonious model) in the first plot; the second plot shows what happens when the smoothing parameter is set to zero, so the penalty has no effect (interpolation); the right plot shows when the smoothing parameter is set to a very large value, giving a straight line. Smooths of this kind are often referred to as a *basis-penalty smoothers*.
 
 **say something about knots!**
 
+**GLS: Given that I just dropped in the quadratic form of the penalty do we want to explain this, mention $\mathbf{S}$, etc?**
 
 **say something about penalty matrices and also smoothing parameters**
 
-```{r lambda, echo=FALSE, message=FALSE, results='hide', fig.width=6, fig.height=3, cache=TRUE, fig.cap="\\label{fig:smoothing_effect}Examples of how different choices of the smoothing parameter effect the resulting function. Data (points) were generated from the blue function and noise added to them. In the left plot the smoothing parameter was estimated to give a good fit to the data, in the middle plot the smoothing parameter was set to zero, so the penalty has no effect and the function interpolates the data, the right plot shows when the smoothing parameter is set to a very large value, so the penalty removes all terms that have any wigglyness, giving a straight line. Numbers in the $y$ axis labels show the estimated degrees of freedom for the term.", messages=FALSE, dev=c('pdf')}
+```{r lambda, echo=FALSE, message=FALSE, results='hide', fig.width=8, fig.height=2.5, cache=TRUE, fig.cap="\\label{fig:smoothing_effect}Examples of how different choices of the smoothing parameter effect the resulting function. Data (points) were generated from the blue function and noise added to them. In the left plot the smoothing parameter was estimated to give a good fit to the data, in the middle plot the smoothing parameter was set to zero, so the penalty has no effect and the function interpolates the data, the right plot shows when the smoothing parameter is set to a very large value, so the penalty removes all terms that have any wigglyness, giving a straight line. Numbers in the $y$ axis labels show the estimated degrees of freedom for the term.", messages=FALSE, dev=c('pdf'), out.width="\\linewidth"}
 # example of varying lambda
 
 library(mgcv)
+library("ggplot2")
+library("cowplot")
```

Comment:
I would prefer to load *cowplot* and then set the theme. The problem was that when you were writing the separate sections of the paper as standalone Rmd's (or even scripts?) you made chunks standalone, so *cowplot* was being loaded in multiple places and in some instances you were explicitly adding the cowplot theme on individual plots.

Because of how I did the edits to separate sections in separate branches, I wasn't seeing the results of my changes to the plot code when compiling the full document.

I'll fix this once we're happy with my changes and these PRs get merged. We can have these common packages loaded in `full_document.Rmd` and I can reset the theme immediately after loading *cowplot* there.

There are quite a few places where we load packages many times &mdash; this isn't a problem but it is trickier to see what you need to repeat the analysis/follow along/etc.

Reply:
Ya, I will be the first to confess that I was sloppy on the first draft of this; part of this arose from writing it as separate sections, and part of it from me not being as careful as possible when coding. 

I think your suggested approach, of loading packages once in full_document.Rmd, is great, and should really clarify things. I can take a look through at some point after you have a run through to make sure that all the dependencies are as clear as possible.